numalogic/udfs/tools.py

```diff
@@ -250,7 +268,7 @@ def __construct_train_key(keys: KEYS) -> str:
     def __fetch_ts(self, key: str) -> _DedupMetadata:
         try:
             data = self.client.hgetall(key)
-        except RedisError:
+        except Exception:
```

Comment:
Any specific reason for this?

Reply:
Yes there is an importError that happens when the user does not install Redis as an optional dependency.